Lib/zipfile.py

```diff
@@ -2228,11 +2229,18 @@ def joinpath(self, add):
 
     @staticmethod
     def _add_implied_dirs(names):
-        return names + [
+        subdirs = set([
             name + "/"
             for name in map(posixpath.dirname, names)
             if name and name + "/" not in names
-        ]
+        ])
+        missing_dirs = set()
```

Comment:
I prefer the pattern used in the test file, where the set was constructed in a single expression instead of an init/for/append pattern.

Reply:
done